### PR TITLE
Profile editor: Integrate cutoff

### PR DESCRIPTION
**Pull request by larsjohnsen**
New look:
untitled

### CODE REVIEW

File: src/UI/Settings/Profile/Edit/EditProfileItemView.js [hunk 1]
```diff
@@ -1,5 +1,21 @@
 var Marionette = require('marionette');
 
 module.exports = Marionette.ItemView.extend({
-    template : 'Settings/Profile/Edit/EditProfileItemViewTemplate'
-});
\ No newline at end of file
+    template : 'Settings/Profile/Edit/EditProfileItemViewTemplate',
+    ui: {
+        cutoff: "input[name=cutoff]"
```
Comment (markus101) [inline src/UI/Settings/Profile/Edit/EditProfileItemView.js]: Use an x-class to select UI elements.

File: src/UI/Settings/Profile/profile.less [hunk 2]
```diff
@@ -30,6 +30,33 @@
   }
 }
 
+ul.ui-sortable {
```
Comment (markus101) [inline src/UI/Settings/Profile/profile.less]: Follow the existing style, 2 spaces for indents.

File: src/UI/Settings/Profile/profile.less [hunk 3]
```diff
@@ -30,6 +30,33 @@
...
+    }
+
+    .selected {
+        .cutoff {
+            display: inline-block;
+            margin-top: 2px;
+            margin-right: 5px;
+            width: 90px;
+
+            input[name=cutoff] {
```
Comment (markus101) [inline src/UI/Settings/Profile/profile.less]: Use a class for applying styles.

File: src/UI/Settings/Profile/profile.less [hunk 4]
```diff
@@ -30,6 +30,33 @@
...
+                &:not(:checked)~[notSelected] {
```
Comment (markus101) [inline src/UI/Settings/Profile/profile.less]: Why do you need these two conditions, its always `display: block`

File: src/UI/Settings/Profile/profile.less [hunk 5]
```diff
@@ -30,6 +30,33 @@
   }
 }
 
+ul.ui-sortable {
+    width: 100%;
+
+    .cutoff > * {
```
Comment (markus101) [inline src/UI/Settings/Profile/profile.less]: This is a pretty aggressive selector, something more specific should be used.

File: src/UI/Settings/Profile/Edit/EditProfileItemView.js [hunk 6]
```diff
@@ -1,5 +1,21 @@
...
+    },
+    onRender: function() {
+        if (this.model.get("quality").id === this.collectionListView.model.get("cutoff").id) {
+            this.ui.cutoff.prop("checked", true);
+        }
+    },
+    events : {
+        'click .cutoff': function($event) {
```
Comment (markus101) [inline src/UI/Settings/Profile/Edit/EditProfileItemView.js]: Use a named function to handle the click. We use `e` as the event handler.

File: src/UI/Settings/Profile/Edit/EditProfileItemView.js [hunk 7]
```diff
@@ -1,5 +1,21 @@
...
+            this.collectionListView.model.set("cutoff", this.model.get("quality"));
+
+            // Do not let event bubble, as the parent will register the click as select / deselect
```
Comment (markus101) [inline src/UI/Settings/Profile/Edit/EditProfileItemView.js]: This is one thing I don't like about this change, if you're off the mark you toggle the quality off completely.

**Comment by markus101** [comment]
At a minimum this needs validation to ensure you can't have the cutoff set to a quality thats not wanted. Looks like we need had server-side validation for that, but should.

I'm not sure I like this change though, it makes the UI feel cluttered.

**Comment by Taloth** [comment]
I have to agree with markus on the UI look.

When I thought about it earlier I considered creating a 'slider' drag-able control that would highlight the area in which we still perform upgrades.
Kinda like:
image

Just a mockup of course. Needs other colors, text is wrong too.

We wanted to get rid of the term cutoff coz it isn't particularly obvious to the user what it means. Visualizing it is probably better.

**Comment by larsjohnsen** [comment]
Taltoth: I like your idea better. 

An alternative that would require adding a dummy item to the list:
image

**Comment by thezoggy** [comment]
leave cutoff as a drop down, and upon selection visually show where the cutt off is by the bg color change... so they see where it would stop. i think the biggest confusion with the list of qualities is the unknown of which way (top to bottom or bottom to top) does the quality get replaced/matter.

**Pull request closed without merging**